pydecred/tests.py

```diff
@@ -0,0 +1,2018 @@
+
+                # Encode again and compare against the original.
+                encoded = decoded.address()
+                self.assertEqual(test.encoded, encoded)
+
+                # Perform type-specific calculations.
+                if isinstance(decoded, crypto.AddressPubKeyHash):
+                    d = ByteArray(b58decode(encoded))
+                    saddr = d[2 : 2+crypto.RIPEMD160_SIZE]
+
+                elif isinstance(decoded, crypto.AddressScriptHash):
+                    d = ByteArray(b58decode(encoded))
+                    saddr = d[2 : 2+crypto.RIPEMD160_SIZE]
+
+                elif isinstance(decoded, crypto.AddressSecpPubKey):
+                    # Ignore the error here since the script
+                    # address is checked below.
+                    try:
+                        saddr = ByteArray(decoded.string())
+                    except Exception:
+                        saddr = test.saddr
+
+                elif isinstance(decoded, crypto.AddressEdwardsPubKey):
+                    # Ignore the error here since the script
+                    # address is checked below.
+                    # saddr = ByteArray(decoded.String())
+                    self.fail("Edwards sigs unsupported")
+
+                elif isinstance(decoded, crypto.AddressSecSchnorrPubKey):
+                    # Ignore the error here since the script
+                    # address is checked below.
+                    # saddr = ByteArray(decoded.String())
+                    self.fail("Schnorr sigs unsupported")
+
+                # Check script address, as well as the Hash160 method for P2PKH and
+                # P2SH addresses.
+                self.assertEqual(saddr, decoded.scriptAddress(), test.name)
+
+                if isinstance(decoded, crypto.AddressPubKeyHash):
+                    self.assertEqual(decoded.pkHash, saddr)
+
+                if isinstance(decoded, crypto.AddressScriptHash):
+                    self.assertEqual(decoded.hash160(), saddr)
+
+            if not test.valid:
+                # If address is invalid, but a creation function exists,
+                # verify that it returns a nil addr and non-nil error.
+                if test.f != None:
+                    try:
+                        test.f()
+                        self.fail("%s: address is invalid but creating new address succeeded" % test.name)
+                    except Exception:
+                        pass
+                continue
+
+            # Valid test, compare address created with f against expected result.
+            try:
+                addr = test.f()
+            except Exception as e:
+                self.fail("%s: address is valid but creating new address failed with error %s", test.name, e)
+            self.assertEqual(addr.scriptAddress(), test.scriptAddress, test.name)
+
+    def test_extract_script_addrs(self):
+        from tinydecred.pydecred import mainnet
+        scriptVersion = 0
+   
```

Comment:
This is really my first time looking at a python test, but it looks like you need to add:
```
if __name__ == '__main__':
    unittest.main()
```
To the bottom in order to test as per:
https://docs.python.org/3/library/unittest.html

After adding those lines, I got this:
<details close>
<summary>test results</summary>

```
$ python3 pydecred/tests.py 
.Emsg: {'event': 'subscribeResp', 'message': {'success': True, 'request_event': 'address:Dcur2mcGjmENx4DhNqDctW5wJCVyT3Qeqkx', 'request_id': 6, 'data': 'subscribed to address:Dcur2mcGjmENx4DhNqDctW5wJCVyT3Qeqkx'}}
msg: {'done': 'done'}
. no stake pool credentials provided. skipping stake pool test
s no stake pool credentials provided. skipping stake pool test
s no stake pool credentials provided. skipping stake pool test
s...............
======================================================================
ERROR: test_purchase_ticket (__main__.TestDcrdata)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "pydecred/tests.py", line 1974, in test_purchase_ticket
    ticketPrice = self.stakeDiff()
AttributeError: 'TestDcrdata' object has no attribute 'stakeDiff'

----------------------------------------------------------------------
Ran 21 tests in 10.503s

FAILED (errors=1, skipped=3)
```
</details>

Reply:
I typically test by specifying the module, e.g.

`python -m unittest crypto`

Looks like you're looking into streamlining testing in #18, so I'll let you find the best solution for our needs.

Reply:
`stakeDiff` error fixed.